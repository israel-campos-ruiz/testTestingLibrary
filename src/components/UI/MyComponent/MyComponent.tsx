import { FC, } from 'react';
interface Props {
    label:string,
    defaultValue:string,
    isDisable:boolean,
    onChange?:any
}

const MyComponent:FC<Props> = ({label,onChange, defaultValue, isDisable}) => {

  return (
    <>
      <label htmlFor="name" aria-label={label}>
       {label}
      </label>
      <input
        onChange={onChange}
        id="inputTextTest"
        type="text"
        name="name"
        defaultValue={defaultValue}
        disabled={isDisable}
    
      />
    </>
  );
};

export default MyComponent;
