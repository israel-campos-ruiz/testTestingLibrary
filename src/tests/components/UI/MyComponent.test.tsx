/* eslint-disable testing-library/no-node-access */
import userEvent from '@testing-library/user-event'
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import MyComponent from '../../../components/UI/MyComponent/MyComponent';

describe("testing in <MyComponent/>", () => {
    const mockfn = jest.fn()
    const initialProps = {
        label:'soyInput',
        onchange:mockfn,
        defaultValue:'',
        isDisable:false,
    }
  const renderMyComponent = (props = initialProps) =>
    render(
  
        <BrowserRouter>
        <MyComponent {...props} />
        </BrowserRouter>
    );

  it("Should match with snapshoot", () => {
    const {asFragment} = renderMyComponent();
    const initialRender = asFragment();
    expect(initialRender).toMatchSnapshot();
  });

  it("Should be render", () => {
    const { asFragment } = renderMyComponent();
    const initialRender = asFragment();
    expect(initialRender).not.toBeNull();
  });


  it("Should exist a label", () => {
    renderMyComponent();
    //bad way
    expect(!!screen.getByLabelText("soyInput")).toBe(true)
    // correct way
    expect(screen.getByLabelText('soyInput')).toBeInTheDocument()
  });

  it("Should be disable", () => {
    renderMyComponent({...initialProps, isDisable:true});
    const { defaultValue } = initialProps
    const view = screen.getByDisplayValue(defaultValue)
    expect(view).toBeDisabled()
  })
  it("Should be enable", () => {
    renderMyComponent({...initialProps, isDisable:false});
    const { defaultValue } = initialProps
    const view = screen.getByDisplayValue(defaultValue)
    expect(view).not.toBeDisabled()
  })


 

  it("Should exist a input", () => {
    renderMyComponent();
   const input =  document.getElementById('inputTextTest');
   const {label} = initialProps
   //bad way
   expect(!!input).toBe(true)
   //correct way 
  expect(screen.getByLabelText(label)).toBeInTheDocument()

  })

  it('Should be able to type', async () => {
      
    renderMyComponent();
    const { defaultValue,} = initialProps
    const input = screen.getByDisplayValue(defaultValue)
    userEvent.type(input, 'typing')  
    expect(input).toHaveValue('typing')
  
  });

  it('Should onChange called',  () => {
    renderMyComponent();
    const { defaultValue, onchange } = initialProps
    const input = screen.getByDisplayValue(defaultValue)
    userEvent.clear(input)
    userEvent.type(input, 'typing') 
    expect(onchange).toHaveBeenCalled()
  });

});
