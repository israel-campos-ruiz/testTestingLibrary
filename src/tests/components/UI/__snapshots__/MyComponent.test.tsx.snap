// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing in <MyComponent/> should match with snapshoot 1`] = `
<DocumentFragment>
  <label
    aria-label="soyInput"
    for="name"
  >
    soyInput
  </label>
  <input
    id="inputTextTest"
    name="name"
    type="text"
    value=""
  />
</DocumentFragment>
`;
